activerecord/lib/active_record/schema_migration.rb

```diff
@@ -34,10 +34,8 @@ def delete_version(version)
     end
 
     def delete_all_versions
-      @pool.with_connection do |connection|
```

Comment:
This one is to avoid checking in / out many times. It's basically batching.

Reply:
Hmm, even acknowledging that this is only a helper for our own tests... I was going to suggest we could add a comment here, but: what if we were to shove an `unless version == :all` onto line 29 above? 🤔

Reply:
Hum, right, it maybe doesn't matter much here.

Reply:
> This one is to avoid checking in / out many times. It's basically batching.

Yeah, right. I added that back and added a comment.